Content pack TypeScript config needs NodeNext module resolution for generated imports

## Problem

When a content pack's generated code imports from `.generated.js` files using explicit `.js` extensions, TypeScript compilation fails without `NodeNext` module resolution configuration.

## Root Cause

The content compiler generates TypeScript modules with ESM-compatible imports that use explicit `.js` extensions:

```typescript
// src/index.ts in a content pack
import {
  PACK__U40_IDLE_U2D_ENGINE_U2F_SAMPLE_U2D_PACK as SAMPLE_PACK,
  // ...
} from './generated/@idle-engine/sample-pack.generated.js';
```

This pattern is required for ESM compatibility - TypeScript source files use `.js` extensions in imports even though the source files are `.ts`. The JavaScript runtime needs `.js` extensions, and TypeScript must be configured to understand this mapping.

The base `tsconfig.base.json` uses standard Node module resolution:

```json
// tsconfig.base.json
{
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "Node"
  }
}
```

The `Node` module resolution strategy does not support resolving `.js` imports to `.ts` source files.

## Expected Behavior

TypeScript should successfully compile content packs that import from generated `.js` modules, resolving them to the corresponding `.ts` source files.

## Actual Behavior

```
error TS2307: Cannot find module './generated/@idle-engine/sample-pack.generated.js' 
or its corresponding type declarations.
```

The TypeScript compiler cannot locate `sample-pack.generated.ts` when the import specifies `.js`.

## Impact

- **Build failure**: Content packs fail to compile with `tsc`
- **Complete blocker**: Cannot use content compiler output without correct configuration
- **Affects all content packs**: Any package importing generated modules needs this fix

## Recommended Solution

Add `NodeNext` module resolution to content pack `tsconfig.json`:

```json
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "dist",
    "rootDir": "src",
    "declaration": true,
    "declarationMap": true,
    "sourceMap": true,
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "types": ["node"]
  },
  "include": ["src/**/*"],
  "exclude": ["dist", "node_modules", "src/**/*.test.ts"]
}
```

Key settings explained:
- `"module": "NodeNext"` - Use Node.js ESM module system semantics
- `"moduleResolution": "NodeNext"` - Resolve `.js` imports to `.ts` files per Node.js ESM rules
- `"types": ["node"]` - Include Node.js type definitions (required when overriding base config)

Reference implementation: `/packages/content-sample/tsconfig.json`

## Acceptance Criteria

- [ ] Documentation for creating content packs requires `NodeNext` module resolution
- [ ] The reason for `NodeNext` is explained: ESM compatibility with explicit `.js` extensions
- [ ] Content pack creation guide references `content-sample` as the canonical example
- [ ] The relationship between content compiler output and module resolution is documented

## Technical Background

### Why `.js` Extensions in TypeScript?

Node.js ESM requires explicit file extensions in import statements. When TypeScript compiles `import { x } from './foo.js'`, it outputs the same path unchanged. The `.js` file exists at runtime after compilation.

During compilation, TypeScript with `NodeNext` resolution understands that `./foo.js` should resolve to `./foo.ts` in the source tree.

### Why Not Use the Base Config?

The base `tsconfig.base.json` uses `"moduleResolution": "Node"` which:
- Does not require file extensions
- Cannot resolve `.js` to `.ts`
- Is designed for CommonJS or bundler-based workflows

Content packs use unbundled ESM and must use `NodeNext` for compatibility with the content compiler's generated output.

## Related Files

- `/packages/content-sample/tsconfig.json` - Reference implementation with `NodeNext`
- `/packages/content-sample/src/index.ts` - Example of `.js` import pattern (line 12)
- `/packages/content-compiler/src/artifacts/module.ts` - Generates the TypeScript module with imports
- `/tsconfig.base.json` - Base config using `Node` resolution (must be overridden)

--- a/docs/content-dsl-usage-guidelines.md
+++ b/docs/content-dsl-usage-guidelines.md
@@ -45,10 +45,10 @@ Start from the structure maintained in `packages/content-sample`:
 
 ### TypeScript Configuration
 
-Content packs must include specific TypeScript declaration settings in their
-`tsconfig.json` to ensure proper `.d.ts` file generation. Without these
-settings, the `"types"` field in `package.json` will point to non-existent
-files, breaking type checking for consumers.
+Content packs must include specific TypeScript settings in their `tsconfig.json`
+to ensure proper module resolution and `.d.ts` file generation. Without these
+settings, TypeScript compilation will fail or the `"types"` field in
+`package.json` will point to non-existent files.
 
 **Required settings:**
 
@@ -57,6 +57,9 @@ files, breaking type checking for consumers.
 | `declaration` | `true` | Generates `.d.ts` type declaration files in `dist/` |
 | `declarationMap` | `true` | Enables "Go to Definition" to navigate to source `.ts` files |
 | `sourceMap` | `true` | Enables debugging with original TypeScript source |
+| `module` | `NodeNext` | Use Node.js ESM module system semantics |
+| `moduleResolution` | `NodeNext` | Resolve `.js` imports to `.ts` files per Node.js ESM rules |
+| `types` | `["node"]` | Include Node.js type definitions (required when overriding base config) |
 
 **Recommended `tsconfig.json` template:**
 
@@ -68,17 +71,28 @@ files, breaking type checking for consumers.
     "rootDir": "src",
     "declaration": true,
     "declarationMap": true,
-    "sourceMap": true
+    "sourceMap": true,
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "types": ["node"]
   },
   "include": ["src/**/*"],
   "exclude": ["dist", "node_modules", "src/**/*.test.ts"]
 }
 ```
 
+> **Why `NodeNext` module resolution is required**: The content compiler
+> generates TypeScript modules that use ESM-compatible imports with explicit
+> `.js` extensions (e.g., `import { ... } from './generated/pack.generated.js'`).
+> The base config uses `"moduleResolution": "Node"`, which cannot resolve these
+> `.js` imports to their corresponding `.ts` source files during compilation.
+> Content packs must override with `NodeNext` to enable this resolution.
+>
 > **Why these settings are not in `tsconfig.base.json`**: The base config
 > intentionally omits declaration settings because not all packages require
-> them. Content packs must explicitly enable declarations to generate the
-> `.d.ts` files that consuming packages depend on.
+> them. It uses `Node` module resolution because not all packages use ESM
+> imports with `.js` extensions. Content packs must explicitly configure both
+> to work with the content compiler's generated output.
 
 See `packages/content-sample/tsconfig.json` for the canonical reference
 implementation.
@@ -123,9 +137,10 @@ documentation.
 - [ ] Create a new workspace package (for example `packages/<pack-slug>`) with a
   `content/` directory that mirrors the layout in
   `packages/content-sample/README.md`.
-- [ ] Configure `tsconfig.json` with required declaration settings (`declaration`,
-  `declarationMap`, `sourceMap`)—see the TypeScript Configuration section above
-  or copy from `packages/content-sample/tsconfig.json`.
+- [ ] Configure `tsconfig.json` with required module resolution and declaration
+  settings (`module`, `moduleResolution`, `types`, `declaration`, `declarationMap`,
+  `sourceMap`)—see the TypeScript Configuration section above or copy from
+  `packages/content-sample/tsconfig.json`.
 - [ ] Create `eslint.config.js` with `restrictCoreInternals: 'error'`—see the
   ESLint Configuration section above or copy from
   `packages/content-sample/eslint.config.js`.

--- a/docs/content-quick-reference.md
+++ b/docs/content-quick-reference.md
@@ -8,17 +8,25 @@ Use this as a fast lookup. For narrative guidance and full examples, see
 
 ## Required tsconfig.json settings
 
-Content packs must include these settings in `tsconfig.json` for proper type exports:
+Content packs must include these settings in `tsconfig.json` for proper module
+resolution and type exports:
 
 ```json
 {
   "compilerOptions": {
     "declaration": true,
     "declarationMap": true,
-    "sourceMap": true
+    "sourceMap": true,
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "types": ["node"]
   }
 }
 ```
+
+> **Why `NodeNext`?** The content compiler generates ESM-compatible imports with
+> explicit `.js` extensions. `NodeNext` module resolution allows TypeScript to
+> resolve these `.js` imports to their corresponding `.ts` source files.
 
 See `packages/content-sample/tsconfig.json` for the complete template.
 

